### PR TITLE
Add Step Progress Component

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

**Motivation:** Add a component as part of the Jetpack Server Credentials Project

* Add StepProgress component
* Add StepProgressExample Component

Screen Shot 2020-09-01 at 3 27 54 PM

#### Testing instructions

1. Navigate to `/devdocs/design/step-progress` on branch
2. Confirm the steps are swapped to primary color as they are reached

### CODE REVIEW

File: client/components/step-progress/index.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+import './style.scss';
+
+interface Props {
+	steps: string[];
+	currentStep: number;
+}
+
+const StepProgress: FunctionComponent< Props > = ( { steps, currentStep } ) => {
+	const getElementNumberClass = ( stepNumber: number ): boolean => {
```
Comment (rcanepa) [inline client/components/step-progress/index.tsx:17]: I think the return type of this function is `string`.

File: client/components/step-progress/index.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		if ( currentStep === stepNumber ) {
+			return 'step-progress__element-number-current';
+		}
+		return stepNumber < currentStep
+			? 'step-progress__element-number-complete'
+			: 'step-progress__element-number';
+	};
+
+	const getStepNameClass = ( stepNumber: number ): boolean => {
```
Comment (rcanepa) [inline client/components/step-progress/index.tsx:26]: Same here.

Comment (cbauerman) [reply client/components/step-progress/index.tsx:26]: copy and paste strikes again!

**Comment by cbauerman** [comment]
> LGTM. I can definitely see it in `devdocs/design` now.

for the record all example components need to have the `displayName` property set!

**Pull request merged**